great_expectations/rule_based_profiler/domain_builder/column_domain_builder.py

```diff
@@ -131,9 +179,225 @@ def _get_domains(
         """
         Obtains and returns domains for all columns of a table (or for configured columns, if they exist in the table).
         """
-        return build_simple_domains_from_column_names(
-            column_names=self.get_effective_column_names(
-                variables=variables,
-            ),
-            domain_type=self.domain_type,
+        batch_ids: List[str] = self.get_batch_ids(variables=variables)
```

Comment:
Not blocking but this method is getting pretty large, could be split up in a future refactor.